diagnostics-app/src/app/app.component.it.spec.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/**
+ * @fileoverview Test file for the app.component.ts
+ */
+
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { AppComponent } from './app.component';
+import { Theme } from 'src/app/core/types/global.types';
+import { ThemeService } from './core/services/theme.service';
+
+describe('component: app', () => {
+  let fixture: ComponentFixture<AppComponent>;
+  let component: AppComponent;
+  let themeService: ThemeService;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [RouterTestingModule],
+      declarations: [AppComponent],
+      schemas: [NO_ERRORS_SCHEMA],
+      providers: [ThemeService],
+    }).compileComponents();
+  });
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(AppComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+    themeService = TestBed.inject(ThemeService);
+  });
+
+  it('set cssClass on receiving update from themeService', () => {
+    expect(component.cssClass).toBe('dark');
+    let currentClass: Theme;
+
+    // subscribe to changes from themeService
+    // expect statements will run on receiving toggleUpdates
+    themeService.themeChanged$.subscribe((next: Theme) => {
+      if (currentClass === 'dark') {
+        // gets executed when currentClass is dark
+        expect(next).toBe('light');
+      } else {
+        // gets executed when currentClass is light
+        expect(next).toBe('dark');
+      }
+    });
+
+    // sets current class 'dark'
+    currentClass = component.cssClass;
+    // trigger manually to simulate toggleTheme method from some component
+    themeService.toggleTheme();
+    fixture.detectChanges();
+    // class should change to light
+    expect(component.cssClass).toBe('light');
+
+    // sets current class 'light'
+    currentClass = component.cssClass;
+    themeService.toggleTheme();
+    fixture.detectChanges();
+    // class should change to dark
+    expect(component.cssClass).toBe('dark');
```

Comment:
we don't need to toggleTheme() twice in the same test. It is enough to test that toggleTheme() "works"